frontend/src/components/Explorer.vue

```diff
@@ -110,10 +110,18 @@ export default {
       model: state => state.models.model,
     }),
   },
+  mounted() {
+    const modelShortName = this.$route.params.model;
+    if (modelShortName && this.model && modelShortName !== this.model.short_name) {
+      this.selectModel(modelShortName);
+    }
+  },
   methods: {
-    async selectModel(modelShortName) {
-      if (modelShortName !== this.model.short_name) {
+    selectModel(modelShortName) {
+      if (modelShortName !== this.$route.params.model) {
```

Comment:
This check is a bit weird here in the context looking how on line 114
```
const modelShortName = this.$route.params.model;
```

I'm really not sure how to improve this by reducing all the checks, especially that this setup seems to work.

Reply:
I'm not sure either since these two `if` conditions are needed depending on the conditions and where `selectModel` is being called from. We could perhaps write two different methods instead, such as `selectModel` and `restoredModelFromURL` but since this method is not overcomplicated at the moment I think we can hold off on that.

Reply:
I think some refactoring would be due, since the last check on line 115 is actually duplicated in 124.

Reply:
Ah you are right! I was looking at line 114 first but we could indeed remove the second condition in line 115. Since this branch is already merged I will include this in a different PR.